common/types/metrics.ts

```diff
@@ -14,9 +15,10 @@ export interface MetricData {
 export interface MetricType extends VisualizationType {
   id: string;
   savedVisualizationId: string;
+  query: string;
   x: number;
   y: number;
   w: number;
   h: number;
-  metricType: 'savedCustomMetric' | 'prometheusMetric';
+  catalog: string;
```

Comment:
This logic around choosing if a record is "customMetric" or "prometheus" is simplified to just knowinng what catalog it is from.